File: 01_Programming_Basics/03_Programming_Basics_with_JavaScript/05_While_Loop/03_GraduationPt2.js
function GraduationPt2(args) {
  let name = args[0];
  let i = 1;
  let n = parseFloat(args[i]);
  let nTotal = 0.0;
  let year = 1;

  let excluded = false;
  let excludedTimes = 0;

  while(year <= 12) {

    if (n >= 4.00) {
      nTotal += n;
      year += 1;
    } else {
      excludedTimes++;

      if (excludedTimes == 2) {excluded = true; break;}
    }

    i += 1;
    n = parseFloat(args[i]);
  }

  if (excluded == true) {
    console.log(name + " has been excluded at " + year + " grade");
  } 
  else {
    let average = nTotal / 12;
    console.log(name + " graduated. Average grade: " + average.toFixed(2));
  }
}

File: 01_Programming_Basics/01_Programming_Basics_with_C++/08_While_Loop_Lab/03_Graduation_Pt2.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main() {
  string name = "";
  double grade = 0;
  double sumGrade = 0;
  double counter = 0;
  double counterExcluded = 0;
  bool validateExcluded = false;

  cin >> name;
  
  while (true) {
    if (counter == 12) { break; }  
    if (counterExcluded == 2) { validateExcluded = true; break; }

    cin >> grade;

    if (grade >= 4) {
      sumGrade += grade;
      counter++;
    } else {
      counterExcluded++;
    }

  }

  if (validateExcluded == true) {
    cout << name << " has been excluded at " << counter + 1 << " grade" << endl;
  } else {
    cout << name << " graduated. Average grade: " << fixed << setprecision(2) << sumGrade / 12 << endl;     
  }

  return 0;

}